components/event-processor-manager/org.wso2.carbon.event.processor.manager.core/src/main/java/org/wso2/carbon/event/processor/manager/core/internal/CarbonEventManagementService.java

```diff
@@ -393,4 +399,29 @@ private void checkMemberUpdate() {
             }
         }
     }
+
+    @Override
+    public void updateLatestEventSentTime(String publisherName, int tenantId, long timestamp){
+
+        stormEventPublisherSyncMap.putAsync(tenantId + "-" + publisherName,
+                EventManagementServiceValueHolder.getHazelcastInstance().getCluster().getClusterTime());
+    }
+
+    @Override
+    public long getLatestEventSentTime(String publisherName, int tenantId){
+        if(stormEventPublisherSyncMap == null){
+            stormEventPublisherSyncMap = EventManagementServiceValueHolder.getHazelcastInstance()
+                    .getMap(ConfigurationConstants.STORM_EVENT_PUBLISHER_SYNC_MAP);
+        }
+        Object latestTimePublished = stormEventPublisherSyncMap.get(tenantId + "-" + publisherName);
+        if (latestTimePublished != null) {
+            return (Long)latestTimePublished;
```

Comment:
Thanks @nirmal070125, Since the stormEventPublisherSyncMap is already a generic map, I changed the latestTimePublished variable type to Long in the pull request - https://github.com/wso2/carbon-analytics-common/pull/121